src/Features/LanguageServer/Protocol/Handler/Configuration/DidChangeConfigurationNotificationHandler.cs

```diff
@@ -0,0 +1,212 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Configuration
+{
+    [Method(Methods.WorkspaceDidChangeConfigurationName)]
+    internal partial class DidChangeConfigurationNotificationHandler : ILspServiceNotificationHandler<LSP.DidChangeConfigurationParams>, IOnInitialized
+    {
+        private readonly ILspLogger _lspLogger;
+        private readonly IGlobalOptionService _globalOptionService;
+        private readonly IClientLanguageServerManager _clientLanguageServerManager;
+        private readonly IAsynchronousOperationListener _asynchronousOperationListener;
+        private readonly Guid _registrationId;
+
+        /// <summary>
+        /// All the global <see cref="ConfigurationItem.Section"/> needs to be refreshed from the client. 
+        /// </summary>
+        private readonly ImmutableArray<ConfigurationItem> _configurationItems;
+        private static readonly ImmutableDictionary<string, string> s_languageNameToPrefix = ImmutableDictionary<string, string>.Empty
+            .Add(LanguageNames.CSharp, "csharp")
+            .Add(LanguageNames.VisualBasic, "visual_basic");
+
+        public static readonly ImmutableArray<string> SupportedLanguages = ImmutableArray.Create(LanguageNames.CSharp, LanguageNames.VisualBasic);
+
+        public DidChangeConfigurationNotificationHandler(
+            ILspLogger logger,
+            IGlobalOptionService globalOptionService,
+            IClientLanguageServerManager clientLanguageServerManager,
+            IAsynchronousOperationListener asynchronousOperationListener)
+        {
+            _lspLogger = logger;
+            _globalOptionService = globalOptionService;
+            _clientLanguageServerManager = clientLanguageServerManager;
+            _registrationId = Guid.NewGuid();
+            _asynchronousOperationListener = asynchronousOperationListener;
+            _configurationItems = GenerateGlobalConfigurationItems();
+        }
+
+        public bool MutatesSolutionState => true;
+
+        public bool RequiresLSPSolution => false;
+
+        public async Task HandleNotificationAsync(DidChangeConfigurationParams request, RequestContext requestContext, CancellationToken cancellationToken)
+        {
+            using var _ = _asynchronousOperationListener.BeginAsyncOperation(Methods.WorkspaceDidChangeConfigurationName);
```

Comment:
I *think* the async operation listener isn't needed here - in the test you would just await the call to the did change configuration request.  

This is what we do for other 'notification' style requests too - https://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures.Test.Utilities/LanguageServer/AbstractLanguageServerProtocolTests.cs,610

The only difference being in notifications the client doesn't have to wait for a response (but we can if we want to in unit tests)